docs/docs/migration-guide.mdx

```diff
@@ -126,6 +126,152 @@ confidences. This change can affect the maximum confidence predicted by the
+To migrate automatically to the 3.0 domain format, you can run the following command:
+
+```bash
+rasa data migrate -d DOMAIN --out OUT_PATH
+```
+
+In addition to creating a a valid 3.0 domain in the indicated out path, this command will automatically backup your
+original domain file(s) in a file labeled `original_domain.yml` or `original_domain` directory if a directory was
+provided instead.
+
+Please note that the migrated slot mappings from forms will include mapping conditions for each form.
+To learn more on mapping conditions, you can read more in this [section](./domain.mdx#mapping-conditions).
+
+#### Manually migrating from 2.0 domain format to the 3.0 format
+
+Let's consider the following 2.0 domain file:
+
+```yaml
+entities:
+ - cuisine
+ - number
+ slots:
+   cuisine:
+     type: text
+   num_people:
+     type: float
+   outdoor_seating:
+     type: bool
+ forms:
+   restaurant_form:
+     required_slots:
+         cuisine:
+           - type: from_entity
+             entity: cuisine
+         num_people:
+           - type: from_entity
+             entity: number
+         outdoor_seating:
+           - type: from_intent
+             intent: affirm
+             value: true
+           - type: from_intent
+             intent: deny
+             value: false
+```
+
+Each slot in `slots` will need a new key `mappings` .
+This key is a list of mappings moved from forms, while the `required_slots` field collapses to a list of slot names.
+For example, the initial result of this migration could look like this:
+
+```yaml
+entities:
+ - cuisine
+ - number
+ slots:
+   cuisine:
+     type: text
+     mappings:
+     - type: from_entity
+       entity: cuisine
+   num_people:
+     type: float
+     mappings:
+     - type: from_entity
+       entity: number
+   outdoor_seating:
+     type: bool
+     mappings:
+     - type: from_intent
+       intent: affirm
+       value: true
+     - type: from_intent
+       intent: deny
+       value: false
+ forms:
+   restaurant_form:
+     required_slots:
+	 - cuisine
+     - num_people
+     - outdoor_seating
+```
+
+In case you'd like certain slots to be filled only within the context of a form, you can add mapping conditions like so:
+
+```yaml
+ slots:
+   outdoor_seating:
+     type: bool
+     mappings:
+     - type: from_intent
+       intent: affirm
+       value: true
+       conditions:
+       - active_loop: restaurant_form
+         requested_slot: outdoor_seating
+     - type: from_intent
+       intent: deny
+       value: false
+       conditions:
+       - active_loop: restaurant_form
+         requested_slot: outdoor_seating
+```
+
+Please note it is not required to use mapping conditions for custom slot mappings defined in a custom action inheriting
+from `FormValidationAction`. However if you wish certain custom slot mappings to only be set within the context of a form,
+make sure your custom action inherits from `FormValidationAction` rather than from `ValidationAction`.
```

Comment:
Keep explanation mostly in the main doc and link to it since the point here is domain format, not validation behaviour.